source/common/router/router.cc

```diff
@@ -103,16 +103,13 @@ bool FilterUtility::shouldShadow(const ShadowPolicy& policy, Runtime::Loader& ru
     return false;
   }
 
-  if (policy.defaultValue().numerator() > 0) {
+  if (!policy.runtimeKey().empty()) {
+    // The default is 0/100, which means that the default for a key set but no default value
+    // explicitly set is 0% mirrored.
     return runtime.snapshot().featureEnabled(policy.runtimeKey(), policy.defaultValue(),
                                              stable_random);
   }
 
-  if (!policy.runtimeKey().empty() &&
-      !runtime.snapshot().featureEnabled(policy.runtimeKey(), 0, stable_random, 10000UL)) {
-    return false;
-  }
-
   return true;
```

Comment:
Does this mean if the runtimeKey() is empty, then 100% to mirror the request whatever the default value is?

A little confused about the default value is required field and the runtime key is optional
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#config-core-v3-runtimefractionalpercent

I can't keep the runtime key empty, but I have to set a value for default value, but in the end, it means 100%.